app/lib/system/redis_pool.rb

```diff
@@ -8,8 +8,8 @@ class RedisPool
 
     def initialize(config = {})
       cfg = config.to_h
-      pool_config = cfg.extract!(:size, :pool_timeout)
-      @pool = ConnectionPool.new(size: pool_config[:size] || 5, timeout: pool_config[:pool_timeout] || 5 ) do
+      pool_config = cfg.extract!(:pool_size, :pool_timeout)
+      @pool = ConnectionPool.new(size: pool_config[:pool_size] || 5, timeout: pool_config[:pool_timeout] || 5 ) do
         Redis.new(cfg)
```

Comment:
how does `Redis` know to use `@pool`? Are you sure it's been actually used?

Reply:
I think Redis itself doesn't know whether it's being used in a pooled connection or not. We just use a ConnectionPool instance for Redis accesses from the app, and it's the ConnectionPool itself that creates and maintains the necessary connections.

Well, that's how I understand it at least.

Reply:
Yes, but how `Redis.new(cfg)` knows about the ConnectionPool?

Reply:
In upstream docs I see 
```rb
    @redis ||= ConnectionPool::Wrapper.new do
      Redis.new(url: ENV["REDIS_URL"])
    end
```
I don't know if our implementation is equivalent.

Reply:
> Yes, but how `Redis.new(cfg)` knows about the ConnectionPool?

Does it need to know?...

The `Wrapper` is not exactly equivalent, as I understand it is used to " wrap a single global connection", so no changes are required to the existin code, as [docs](https://github.com/mperham/connection_pool/tree/main?tab=readme-ov-file#migrating-to-a-connection-pool) say.

We use the recommended `ConnectionPool.new { Redis.new }` interface, invoking the Redis's commands through `pool.with { -block- }` : https://github.com/3scale/porta/blob/5b4395980836df430fc6b56c5b32d94513571338/app/lib/system/redis_pool.rb#L25-L27

So, I understand that Redis client doesn't know that it's pooled, only our application knows that.